coordinator/src/trade/mod.rs

```diff
@@ -231,13 +234,44 @@ impl TradeExecutor {
                 .await
                 .context("Failed to open DLC channel")?;
             }
-            TradeAction::OpenSingleFundedChannel => {
+            TradeAction::OpenSingleFundedChannel { external_funding } => {
                 let collateral_reserve_coordinator = params
                     .coordinator_reserve
                     .context("Missing coordinator collateral reserve")?;
-                let collateral_reserve_trader = params
-                    .trader_reserve
-                    .context("Missing trader collateral reserve")?;
```

Comment:
🔧 Shouldn't we calculate the `trader_reserve` before we get here so that we can read it from the `params` like we do with the `coordinator_reserve`?

Reply:
The reason why I am recalculating the trader_reserve is because we can't trust the client on these params, so if we are going the `SingleFundedChannelOpening` path we only, use the funds received via lightning and ignore the params.

Note, I think its also not a good idea to blindly trust the client about the coordinators reserve. We should probably at least verify if the values make sense. I'll leave a FIXME.

Reply:
What I mean is that I would like to recalculate the trader reserve before calling `TradeExecutor::execute`. Then we don't need the `external_funding` to be passed any deeper.

> Note, I think its also not a good idea to blindly trust the client about the coordinators reserve. We should probably at least verify if the values make sense. I'll leave a FIXME.

I agree with that. We could end up locking up more or less funds than desired.

Reply:
> What I mean is that I would like to recalculate the trader reserve before calling `TradeExecutor::execute`. Then we don't need the `external_funding` to be passed any deeper.

Hmm.. that would make the api a bit less convenient as the caller would need to implement the logic for handling the channel opening params before calling execute.

That would mean we'd have to duplicate the logic for "async match". Do you think it's worth it?

Reply:
I guess it's debatable, so none of this is a deal breaker at all.

What I dislike is leaking the knowledge of the external funding any further than necessary, and the fact that the `trader_reserve` field is ignored, when it could be used to carry this information from the outside.

And I think we could avoid code duplication by reusing a helper function.

Reply:
Yeah i think you are right.. I opted to leave this now, since I wanted to go forward with this PR. I think we can address it the next time we touch this part. (which will be probably be very soon :))